### DISCUSSION TITLE
Unable to upload file greater than 2mb

### DISCUSSION THREAD

**Discussion by MUHAMMADSiyab** [livewire/livewire#7610] [1. Help] [answered]
When i try to upload a file lesser than or equal to 2 megabytes, everything goes smooth, but when uploading greater than 2mb, it gives validation error of:

`file failed to upload`

livewire version: ^3.0
laravel version: ^10.10
php version: 8.1

config/livewire.php

```
 'temporary_file_upload' => [
        'disk' => null,    
        'rules' => ['required', 'file', 'max:12288'],      
        'directory' => null,   // Example: 'tmp'                  
        'middleware' => null,  // Example: 'throttle:5,1'         
        'preview_mimes' => [  
            'png', 'gif', 'bmp', 'svg', 'wav', 'mp4',
            'mov', 'avi', 'wmv', 'mp3', 'm4a',
            'jpg', 'jpeg', 'mpga', 'webp', 'wma',
        ],
        'max_upload_time' => 5,
    ];
```

CreateStory.php:

```
#[Rule(['video' => 'mimetypes:video/mp4,video/webm|max:12288'])]
public $video = "";
```

create-story.blade.php:

```
 <input type="file" name="video" id="video" wire:model.lazy="video"           
                    class="hidden" accept="video/*">
```

Also updated the php.ini and increased to post_max_size and upload size to 100mb

**Comment by mcpuishor** [comment]
This does not seem to be a Livewire related issue, but rather PHP configuration.

The uploads are controlled by 2 settings in php.ini

```
upload_max_filesize
 post_max_size
```

Try setting both, ideally with the `post_max_size` higher than the `upload_max_filesize`.

**Reply by MUHAMMADSiyab** [reply]
Changed both settings but still facing the same issue.

**Reply by thehands79** [reply, marked as answer]
Check the values set in phpinfo() both in the webserver and the command line:

```
// you can create a temporary Route like this
Route::get('/phpinfo', function () {
    phpinfo();
});

// from the command line
php -r "phpinfo();" | grep upload_max_filesize
```

I had a similar problem when using [Herd](https://herd.laravel.com/). I tried changing its php.ini values but just couldn't get it to work. In the end I used homebrew to install php 8.2 and configured its php.ini file instead.

**Comment by josegus** [comment]
@MUHAMMADSiyab if you're using Valet and nginx don't forget to change the `client_max_body_size` parameter of Nginx.

https://dorelljames.com/blog/laravel-valet-increase-max-upload-file-size/
